pkg/segment/utils/number_test.go

```diff
@@ -0,0 +1,396 @@
+	assert.NotEqual(t, nil, err)
+
+}
+
+// Test_isFloat64 tests if the number is a float64. It should return set value for Float64 and an error for Int64.
+func Test_isFloat64(t *testing.T) {
+
+	n := Number{}
+	n.SetFloat64(42.0)
+	assert.Equal(t, float64Type, n.ntype())
+
+	val, err := n.Float64()
+	assert.Equal(t, 42.0, val)
+	assert.Equal(t, nil, err)
+
+	_, err = n.Int64()
+	assert.NotEqual(t, nil, err)
+
+}
+
+// Test_reset tests resetting the number. It should return invalidType.
+func Test_reset(t *testing.T) {
+
+	n := Number{}
+	n.SetInt64(42)
+	assert.Equal(t, int64Type, n.ntype())
+	n.Reset()
+	assert.Equal(t, invalidType, n.ntype())
+
+}
+
+// Test_constructor tests the constructor of the number. It should return the correct type.
+func Test_constructor(t *testing.T) {
+
+	n := Number{}
+	assert.Equal(t, invalidType, n.ntype())
+
+}
+
+// Test_numberSetAndGet tests setting and getting the number. It should return the correct value.
+func Test_numberSetAndGet(t *testing.T) {
+	n := Number{}
+
+	// Test Int64
+	n.SetInt64(42)
+	if val, err := n.Int64(); err != nil || val != 42 {
+		assert.Fail(t, "Expected 42, got %v with error %v", val, err)
+	}
+
+	// Test Float64
+	n.SetFloat64(3.14)
+	if val, err := n.Float64(); err != nil || math.Abs(val-3.14) > 1e-6 {
+		assert.Fail(t, "Expected 3.14, got %v with error %v", val, err)
+	}
+
+	// Test Invalid Type
+	n.SetInvalidType()
+	if _, err := n.Int64(); err == nil {
+		assert.Fail(t, "Expected error for invalid type")
+	}
+	if _, err := n.Float64(); err == nil {
+		assert.Fail(t, "Expected error for invalid type")
+	}
+
+	// Test Backfill Type
+	n.SetBackfillType()
+	if val, err := n.Int64(); err != nil || val != 0 {
+		assert.Fail(t, "Expected 0 for backfill, got %v with error %v", val, err)
+	}
+	if val, err := n.Float64(); err != nil || val != 0 {
+		assert.Fail(t, "Expected 0 for backfill, got %v with error %v", val, err)
+	}
+}
+
+// Test_numberConversion tests converting the number. It should return the correct value.
+func TestNumberConversion(t *testing.T) {
+	n := Number{}
+
+	// Test Int64 to Float64 conversion
+	n.SetInt64(42)
+	assert.Equal(t, int64Type, n.ntype())
+	if err := n.ConvertToFloat64(); err != nil {
+		assert.Fail(t, "Conversion failed: %v", err)
+	}
+	if val, err := n.Float64(); err != nil || math.Abs(val-42.0) > 1e-6 {
+		assert.Fail(t, "Expected 42.0, got %v with error %v", val, err)
+	}
+
+	// Test Float64 to Int64 conversion
+	n.SetFloat64(3.14)
+	assert.Equal(t, float64Type, n.ntype())
+	if err := n.ConvertToInt64(); err != nil {
+		assert.Fail(t, "Conversion failed: %v", err)
+	}
+	if val, err := n.Int64(); err != nil || val != 3 {
+		assert.Fail(t, "Expected 3, got %v with error %v", val, err)
+	}
+}
+
+// Test_numberCopyToBuffer tests copying the number to a buffer. It should return the correct value.
+func Test_numberCopyToBuffer(t *testing.T) {
+	n := Number{}
+	n.SetFloat64(3.14)
+	buf := make([]byte, 9)
+	n.CopyToBuffer(buf)
+	if buf[8] != float64Type {
```

Comment:
User `assert.Equal`. Please change any other places where you have code like this.

Reply:
Changed